### DISCUSSION TITLE
plotting multiple graphs

### DISCUSSION THREAD

**Discussion by matiashr** [Blake-Madden/Wisteria-Dataviz#39] [Q&A] [answered]
I want to plot multiple data series in a graph created without import,
but I cant figure out how to add additional graphs, did the following:

dt = std::make_shared<Data::Dataset>();
dt->AddContinuousColumn(L"y");
dt->AddCategoricalColumn(L"x");
dt->Resize(10);
dt->GetCategoricalColumn(L"x")->Fill({ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
dt->GetContinuousColumn(L"y")->Fill({ 8, 2, 9, 7, 9, 89, 97, 82, 68, 92 });
                                        std::shared_ptr<LinePlot> l = std::make_shared<LinePlot>(m_canvas,
                                                                                                std::make_shared<Colors::Schemes::Decade1960s>(),
std::make_shared<IconScheme>(IconScheme{ IconShape::Hexagon}));
                                        l->SetCanvasMargins(5, 5, 5, 5);
                                        l->SetData(dt,  L"y", L"x");
                                        l->GetBottomXAxis().GetTitle().SetText(L"");
                                        l->GetLeftYAxis().GetTitle().SetText(L"");
                                        m_canvas->SetFixedObject(0, 0, l);
                                        m_canvas->SetFixedObject(0, 1, l->CreateLegend(
                                                                LegendOptions().
                                                                IncludeHeader(true).
                                                                PlacementHint(LegendCanvasPlacementHint::RightOfGraph)) );

This works for a single graph, but how do I add more ?

**Comment by Blake-Madden** [comment]
In the demo (`demo/demo.cpp`), look for this section:

```
// Multiple plots
else if (event.GetId() == MyApp::ID_NEW_MULTIPLOT)
```

That will show how to place multiple graphs on a canvas. Basically, you create it with grid (e.g., `m_canvas->SetFixedObjectsGridSize(2, 2);`) and then you can place your graphs on the canvas (in that grid) using `SetFixedObject()`.

**Reply by matiashr** [reply]
That would create one more plot ? , What I mean is add an interpolated plot to the same graph,
I dont really understand this grouping mechanism..

dt->AddContinuousColumn(L"y");
dt->AddContinuousColumn(L"y2");
dt->AddCategoricalColumn(L"x");
dt->Resize(10);
dt->GetCategoricalColumn(L"x")->Fill({ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
dt->GetContinuousColumn(L"y")->Fill({ 8, 2, 9, 7, 9, 89, 97, 82, 68, 92 })
dt->GetContinuousColumn(L"y2")->Fill({ 8, 2, 9, 7, 9, 89, 97, 82, 68, 92 })

**l->SetData(dt, L"y", L"x");** ? this seems to plot x vs y , but I want x vs y, y1

**Comment by Blake-Madden** [comment]
OK, I see.  You would want one continuous column with ALL the Y data, and then a grouping column that would contain the labels Y1 and Y2 to distinguish the Y1 and Y2 data.  And in the X column you would have duplication, where you have a corresponding X value for the Y1 and Y2 values in the Y column.  Basically, a "long" dataset instead of a "wide" one.

There is a dataset pivoting feature in this library that you can your current dataset you design and turn it into the required wide format. I''ll put example of that in here tonight...

**Comment by Blake-Madden** [comment, marked as answer]
If you go into "data/pivot.h", there is a `Pivot` object that you can create. With that, you can call `PivotLonger()` and pass in the dataset that you have constructed and get back a "long" dataset that you can then use with the line plot.

```
Pivot pv;
auto newDataset = pv.PivotLonger(myDataset, { L"x" }, { L"y1", L"y2" }, { L"GROUP" }, L"YValues");
```

At this point, you can pass `newDataset` to the line plot and specify "X" and "YValues" as the X and Y and "GROUP" as the grouping column. From there, is will create a line for the Y1 values and another line for the Y2 values.

**Reply by matiashr** [reply]
Excellent, exactly what I wanted, you should add this to the demo's

**Comment by Blake-Madden** [comment]
This article explains this process of "long" datasets in a bit more detail:

https://tidyr.tidyverse.org/reference/pivot_longer.html

This is basically what this `Pivot::PivotLonger()` is doing.

**Comment by Blake-Madden** [comment]
Did this answer your question?
